sympy/sets/sets.py

```diff
@@ -1427,13 +1427,23 @@ def _contains(self, other):
         return And(*[set.contains(other) for set in self.args])
 
     def __iter__(self):
+        no_iter = True
         for s in self.args:
             if s.is_iterable:
+                no_iter = False
                 other_sets = set(self.args) - set((s,))
                 other = Intersection(other_sets, evaluate=False)
-                return (x for x in s if x in other)
+                for x in s:
+                    c = other.contains(x)
+                    if c == True:
```

Comment:
bool will convert unevaluate `contains` results to True; `fuzzy_bool` can be used but then we are using a function call to convert c to True so we can use `is` to test it. My hunch is that this will be slower than just using the equality check.

My timeit runs show that for `t = S.true`, `fuzzy_bool(t) == True` and `fuzzy_bool(t) is True` essentially take the same time.

So until we do an audit and make sure that all contains routines return S.true, S.false, or an expression, it will be safer to use `== True` to test for S.true or True.